src/led_ble/led_ble.py

```diff
@@ -431,29 +437,64 @@ def _named_effect(self) -> str | None:
+    def _is_hello_fairy(self) -> bool:
+        if self._read_char is None:
+            return False
+        d = self._read_char.descriptors
+        c = d[0].characteristic_uuid if (len(d) > 0) else None
+        return c == HELLO_FAIRY_READ_CHARACTERISTIC
+
     def _notification_handler(self, _sender: int, data: bytearray) -> None:
         """Handle notification responses."""
         _LOGGER.debug("%s: Notification received: %s", self.name, data.hex())
 
-        if len(data) == 4 and data[0] == 0xCC:
-            on = data[1] == 0x23
-            self._state = replace(self._state, power=on)
-            return
-        if len(data) < 11:
-            return
-        model_num = data[1]
-        on = data[2] == 0x23
-        preset_pattern = data[3]
-        mode = data[4]
-        speed = data[5]
-        r = data[6]
-        g = data[7]
-        b = data[8]
-        w = data[9]
-        version = data[10]
-        self._state = LEDBLEState(
-            on, (r, g, b), w, model_num, preset_pattern, mode, speed, version
-        )
+        model_num = 0
+        if self._is_hello_fairy():
+            if data[0] == 0xAA:
+                if data[1] == 0x00:  # hw info
+                    if len(data) > 7:
+                        version_string = data[3:8].decode("ascii")
+                        _LOGGER.debug("version %s", version_string)
+                        self._state = replace(
+                            self._state,
+                            version_num=(data[3] - 48) * 100
+                            + (data[5] - 48) * 10
+                            + (data[7] - 48),
+                        )
+                    if len(data) > 12:
+                        model = data[8:13].decode("ascii")
+                        _LOGGER.debug("model %s", model)
+                    if len(data) > 24:
+                        lights = data[24]  # guessing
+                        _LOGGER.debug("lights %d", lights)
+                    if len(data) > 33:
+                        effects = data[33]  # guessing
+                        _LOGGER.debug("effects %d", effects)
+
+                if data[1] == 0x01:  # state info
+                    if len(data) > 6:
+                        self._state = replace(self._state, power=data[6] > 0)
+        else:
+            if len(data) == 4 and data[0] == 0xCC:
+                on = data[1] == 0x23
+                self._state = replace(self._state, power=on)
+                return
+            if len(data) < 11:
+                return
+            model_num = data[1]
+            on = data[2] == 0x23
+            preset_pattern = data[3]
+            mode = data[4]
+            speed = data[5]
+            r = data[6]
+            g = data[7]
+            b = data[8]
+            w = data[9]
+            version = data[10]
+            self._state = LEDBLEState(
+                on, (r, g, b), w, model_num, preset_pattern, mode, speed, version
+            )
 
```

Comment:
It would be better to abstract this into a new method in the protocol so each protocol can handle this different

Reply:
Agreed.  We don't control the ProtocolBase class though, it comes from flux_led.  One option would be to make a device class to encapsulate the differences between the flux_led devices and the hello fairy devices.  This would include the (send) protocol, this notification_handler, and (to your other comments), the status command, and protocol_for_version_num.

Reply:
Its no problem to merge a PR to flux_led

Reply:
I cleaned up flux_led for Python 3.9+ and released 1.1.0

Reply:
do you want the new protocol method to return LEDBLEState (so move that class into flux-led)?  or a generic dict?